src/h5web/visualizations/nexus/utils.ts

```diff
@@ -106,3 +89,56 @@ export function getNxDataGroup(
 
   return getNxDataGroup(defaultEntity, metadata);
 }
+
+export function getDatasetLabel(
+  dataset: HDF5Dataset,
+  datasetName: string
+): string {
+  const longName = getAttributeValue(dataset, 'long_name');
+  if (longName && typeof longName === 'string') {
+    return longName;
+  }
+
+  const units = getAttributeValue(dataset, 'units');
+  if (units && typeof units === 'string') {
+    return `${datasetName} (${units})`;
+  }
+
+  return datasetName;
+}
+
+export function getDatasetsFromMapping(
+  nxDataGroup: HDF5Group,
+  metadata: HDF5Metadata,
+  nxAxisMapping: (string | undefined)[]
```

Comment:
I would put the mapping as the first argument. Also, I think the function should be named either:

- `getDatasetsFromAxisMapping` (with a parameter named, simply, `mapping`), or
- `getLinkedDatasets` (with a parameter named `names` which would only contain strings, and would therefore be called with `getLinkedDatasets(nxAxisMapping.filter(val => !!val), ...)`.

I reckon the second option might be better, because it keeps the utility function generic. I know it adds a bit of complexity back into the containers, but it remains readable.

Reply:
Settled for `getLinkedDatasets`. I also changed the signature of `getLinkedEntity` so that the `name` is the first argument as well.